gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -67,11 +75,16 @@ public MainViewModel(
 
     // TODO: Move to background service.
     disposeOnClear(
-        projectRepository.getActiveProject().switchMapCompletable(this::syncFeatures).subscribe());
+        projectRepository
+            .getActiveProject()
+            .observeOn(schedulers.io())
+            .switchMapCompletable(this::syncFeatures)
+            .subscribe());
 
     disposeOnClear(
         authenticationManager
             .getSignInState()
+            .observeOn(schedulers.ui())
```

Comment:
`.observeOn(schedulers.ui())` 
Why is this change needed?

Reply:
Our best practices guidelines recommend specifying `observeOn()`  downstream close to `subscribe()`. But I think in this case it's irrelevant to place it here since we only do work inside the `switchMap` operation, which has its own subscription and threads. I'll remove it to avoid confusion.

Reply:
It's needed to ensure we're actually using the UI thread to process values handled by `subscribe()`.